ACPs/194-streaming-asynchronous-execution/README.md

```diff
@@ -40,70 +40,77 @@ This ACP does not introduce these, but some form of asynchronous execution is re
-Under SAE, a block's ordered transactions instead constitute those to be pushed to the queue
-A transaction is said to be _included_ when a block builder adds it to a block and _accepted_ once consensus, performed over blocks, agrees to add it to the queue.
-We refer to the ordered set of enqueued transactions as a _tranche_.
+Under SAE, a block is first _proposed_ by a validator. In order for validators to consider the block valid for consensus, it is verified that _execution_ will later succeed. The block is then _accepted_ by consensus. The act of _accepting_ a block _enqueues_ the block to be _executed_. After the block is _executed_, a following block will reference the execution results and _settle_ all transactions in the _executed_ block.
 
-#### Clearing the queue
+```mermaid
+flowchart LR
+    I[Proposed] --> A[Accepted]
+    A -->|variable delay| E[Executed]
+    E -->|τ seconds| S[Settled]
+    A -. guarantees .-> S
+```
+
+### Block lifecycle
 
-For a block indexed by its height $B$, let its timestamp be $t_B$, measured at per-second resolution.
-We now introduce the concept of a _chunk_ of executed transactions, indexed by its timestamp $t_C$, at the same resolution.
-Whereas proposers _build_ blocks, execution is said to _fill_ chunks and a chunk is considered full when the first of two criteria is met:
+#### Proposing blocks
 
-1. The queue accepted by time $t_C$ is exhausted; i.e. the queue as at the last block with a timestamp $t_B \le t_C$; or
-2. The chunk's remaining gas limit is insufficient for the next transaction in the queue.
+The validator selection mechanism for block production is unchanged. However, block builders are no longer expected to execute transactions during block building.
 
-As there is one chunk per second, the gas limit is simply the gas capacity added per second, $R$, of ACPs 103 and 176.
+The block builder is expected to include transactions based on the most recently settled state and to apply worst-case bounds on the execution of the ancestor blocks prior to the most recently settled block.
 
-A chunk that is filled under criterion (2) may have leftover capacity due to imperfect packing and MUST _donate_ this residual capacity to the next chunk to allow for full utilisation of the chain's gas throughput.
- This is not to say that execution of the queue should be paused until the next second, but that the transaction's allocated chunk may well be in the future.
-For clarification, see the [alternative framing of chunk filling](#alternative-framing-of-chunk-filling).
+The worst-case bounds enforce minimum balances of sender accounts and the maximum required base fee. The worst-case bounds are described [below](#block-validity-and-building).
+
+Prior to adding a proposed block to consensus, all validators MUST verify that the block builder correctly enforced the worst-case bounds while building the block. This guarantees that the block can be executed successfully if it is accepted.
```

Comment:
```suggestion
Prior to proposing a block for consensus, all validators MUST verify that the block builder correctly enforced the worst-case bounds while building the block. This guarantees that the block can be executed successfully if it is accepted.
```

Reply:
I don't think this is quite right. Only one validator really proposes a block (equivalent to building and gossiping a block)

Reply:
updated suggestion

Reply:
Yeah, this suggestion reads as if the block builder and the proposer are different people.